### DISCUSSION TITLE
Onedrive support in Gnome 46

### DISCUSSION THREAD

**Discussion by galettof** [abraunegg/onedrive#2682] [Q&A] [answered]
Hello!

I've been happily using onedrive client for a while and recently saw that the [next version of the Gnome desktop](https://release.gnome.org/46/) is supposed to come with Onedrive support built in. Does anyone here have any insight on how Gnome's Onedrive support might differ in practice from using the onedrive client?

Thanks

**Comment by abraunegg** [comment, marked as answer]
@galettof 
I have not looked at the Gnome 46 capability in any great detail, however my understanding is that the capability will be similar to [onedriver](https://github.com/jstaf/onedriver) and that there will be quite a feature / capability gap between this client (v2.4.x) and an even greater gap with v2.5.x

Without using / testing it, I see these gaps at this stage by reading the code:
* Lack of compliance to key limitations and restrictions. Refer to: https://github.com/abraunegg/onedrive/blob/onedrive-v2.5.0-release-candidate-1/docs/usage.md#guidelines-for-local-file-and-folder-naming-in-the-synchronisation-directory
* Lack of compliance to Microsoft OneDrive's own lack of compliance to POSIX standards
* Lack of capability to support SharePoint sites and all the nuances that comes with SharePoint such as SharePoint modifying your file post upload. Refer to: https://github.com/OneDrive/onedrive-api-docs/issues/935
* Lack of capability to support OneDrive Business Shared Files
* Lack of any client side filtering of any sort
* Lack of any tray icon / progress icon to indicate what is going on - similar to how Windows operates, or https://github.com/bpozdena/OneDriveGUI 

Personally I think this feature may be useful for folk with small OneDrive Personal Accounts, where they wish to utilise the 'files on demand' capability with their system without setting up and using 'onedriver' ... time will tell.

**Reply by galettof** [reply]
Thank you for the detailed response!
